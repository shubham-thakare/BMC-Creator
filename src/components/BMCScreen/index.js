import BMCScreen from './BMCScreen';

export default BMCScreen;
