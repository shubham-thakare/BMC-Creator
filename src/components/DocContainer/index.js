import DocContainer from './DocContainer';

export default DocContainer;
