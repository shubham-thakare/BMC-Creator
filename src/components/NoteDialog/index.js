import NoteDialog from './NoteDialog';

export default NoteDialog;
