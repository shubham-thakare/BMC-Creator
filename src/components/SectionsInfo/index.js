import { KeyPartners } from '../../components/SectionsInfo/KeyPartners';
import { KeyActivities } from '../../components/SectionsInfo/KeyActivities';
import { KeyResources } from '../../components/SectionsInfo/KeyResources';
import { ValuePropositions } from '../../components/SectionsInfo/ValuePropositions';
import { CustomerRelationships } from '../../components/SectionsInfo/CustomerRelationships';
import { Channels } from '../../components/SectionsInfo/Channels';
import { CustomerSegments } from '../../components/SectionsInfo/CustomerSegments';
import { CostStructure } from '../../components/SectionsInfo/CostStructure';
import { RevenueStreams } from '../../components/SectionsInfo/RevenueStreams';

import {
  PopoverBody,
  PopoverQuestion,
  PopoverHintTitle,
  PopoverHintText,
} from './styles/Common.style';

export {
  KeyPartners,
  KeyActivities,
  KeyResources,
  ValuePropositions,
  CustomerRelationships,
  Channels,
  CustomerSegments,
  CostStructure,
  RevenueStreams,
  PopoverBody,
  PopoverQuestion,
  PopoverHintTitle,
  PopoverHintText,
};
