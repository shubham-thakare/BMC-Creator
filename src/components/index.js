import Loading from './Loading';
import SideNav from './SideNav';
import Popover from './Popover';

import {
  KeyPartners,
  KeyActivities,
  KeyResources,
  ValuePropositions,
  CustomerRelationships,
  Channels,
  CustomerSegments,
  CostStructure,
  RevenueStreams,
} from './SectionsInfo';

export {
  Loading,
  SideNav,
  KeyPartners,
  KeyActivities,
  KeyResources,
  ValuePropositions,
  CustomerRelationships,
  Channels,
  CustomerSegments,
  CostStructure,
  RevenueStreams,
  Popover,
};
