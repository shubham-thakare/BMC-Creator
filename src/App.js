import React, { useState } from 'react';
import { BMCScreen, SideNav } from './components';
import { AppContext, contextWithoutState } from './utils/Context';

const App = () => {
  const { Provider } = AppContext;

  const [title, setTitle] = useState('');
  const [version, setVersion] = useState('');
  const [date, setDate] = useState('');
  const [keyPartners, setKeyPartners] = useState({});
  const [keyActivities, setKeyActivities] = useState({});
  const [keyResources, setKeyResources] = useState({});
  const [valuePropositions, setValuePropositions] = useState({});
  const [customerRelationships, setCustomerRelationships] = useState({});
  const [channels, setChannels] = useState({});
  const [customerSegments, setCustomerSegments] = useState({});
  const [costStructure, setCostStructure] = useState({});
  const [revenueStreams, setRevenueStreams] = useState({});

  const contextWithState = {
    ...contextWithoutState,
    title,
    setTitle,
    version,
    setVersion,
    date,
    setDate,
    keyPartners,
    setKeyPartners,
    keyActivities,
    setKeyActivities,
    keyResources,
    setKeyResources,
    valuePropositions,
    setValuePropositions,
    customerRelationships,
    setCustomerRelationships,
    channels,
    setChannels,
    customerSegments,
    setCustomerSegments,
    costStructure,
    setCostStructure,
    revenueStreams,
    setRevenueStreams,
  };

  return (
    <Provider value={contextWithState}>
      <SideNav />
      <BMCScreen />
    </Provider>
  );
};

export default App;
