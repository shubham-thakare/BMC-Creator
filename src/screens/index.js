import BMCScreen from './BMCScreen';

export { BMCScreen };
