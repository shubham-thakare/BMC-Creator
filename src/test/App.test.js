import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';
import { SideNav } from '../components';
import { BMCScreen } from '../screens';

let wrapper;

describe('App Component', () => {
  beforeAll(() => {
    wrapper = shallow(<App />);
  });

  it('To match snapshot of App', async (done) => {
    expect(wrapper).toMatchSnapshot();
    done();
  });

  it('NavBar exists', async (done) => {
    expect(wrapper.find(SideNav)).toHaveLength(1);
    done();
  });

  it('BMCScreen exists', async (done) => {
    expect(wrapper.find(BMCScreen)).toHaveLength(1);
    done();
  });
});
