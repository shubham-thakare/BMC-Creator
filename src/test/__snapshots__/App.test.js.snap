// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component To match snapshot of App 1`] = `
<ContextProviderWrapper>
  <SideNav
    hasFooter={true}
    hasOpenFileOption={true}
    hasSaveFileOption={true}
  />
  <BMCScreen />
</ContextProviderWrapper>
`;
